server/src/main/kotlin/tech/figure/objectstore/gateway/server/ObjectStoreGatewayServer.kt

```diff
@@ -61,4 +76,86 @@ class ObjectStoreGatewayServer(
             responseObserver.onCompleted()
         }
     }
+
+    override fun grantScopePermission(
+        request: GatewayOuterClass.GrantScopePermissionRequest,
+        responseObserver: StreamObserver<GrantScopePermissionResponse>,
+    ) {
+        val requesterAddress = publicKey().getAddress(mainNet = provenanceProperties.mainNet)
+        val grantId = request.grantId.takeIf { it.isNotBlank() }
+        val sourceDetails = "Manual grant request by $requesterAddress for scope ${request.scopeAddress}, grantee ${request.granteeAddress}${if (grantId != null) ", grantId $grantId" else ""}"
+        val grantResponse = scopePermissionsService.processAccessGrant(
+            scopeAddress = request.scopeAddress,
+            granteeAddress = request.granteeAddress,
+            grantSourceAddresses = setOf(requesterAddress),
+            // The application's admin should be able to manually grant any permission that is desired
+            additionalAuthorizedAddresses = setOf(masterKey.publicKey.getAddress(mainNet = provenanceProperties.mainNet)),
+            grantId = grantId,
+            sourceDetails = sourceDetails,
+        )
+        val respond: (accepted: Boolean, granterAddress: String?) -> Unit = { accepted, granterAddress ->
+            responseObserver.onNext(
+                GrantScopePermissionResponse.newBuilder().also { rpcResp ->
+                    rpcResp.request = request
+                    rpcResp.grantAccepted = accepted
+                    granterAddress?.also { rpcResp.granterAddress = it }
+                }.build()
+            )
+            responseObserver.onCompleted()
+        }
+        when (grantResponse) {
+            is GrantResponse.Accepted -> respond(true, grantResponse.granterAddress)
+            is GrantResponse.Rejected -> {
+                logger.warn("REJECTED $sourceDetails: ${grantResponse.message}")
+                respond(false, null)
+            }
+            is GrantResponse.Error -> {
+                logger.error("ERROR $sourceDetails", grantResponse.cause)
+                responseObserver.onError(StatusRuntimeException(Status.UNKNOWN.withDescription(DEFAULT_UNKNOWN_DESCRIPTION)))
+            }
+        }
+    }
+
+    override fun revokeScopePermission(
+        request: GatewayOuterClass.RevokeScopePermissionRequest,
+        responseObserver: StreamObserver<RevokeScopePermissionResponse>,
+    ) {
+        val requesterAddress = publicKey().getAddress(mainNet = provenanceProperties.mainNet)
```

Comment:
there's also an `address()` helper function with this value

Reply:
Oh snap!  Alright, I'll get these updated